docs/community/tutorials-contest/2022/red-dev-ecrecover-x-chain/README.md

```diff
@@ -0,0 +1,366 @@
+
+The xChainECRecoverReadSignature contains the function selector which can be used in solidity smart contracts to execute the precompile.
+
+The getXChainECRecover() function contains the business logic for verifying the signature and returning the X-Chain address.
+
+Please [go take a look at it now](./example/precompile/contract_xchain_ecrecover.go).
+
+Returning the recovered X-Chain address was the trickiest part of writing this code since Avalanche X-Chain addresses are not in the same format as EVM addresses. You will see at the end that we had to return it as a byte[] but pad it with special bytes at the beginning to prevent an error from being thrown as Solidity interprets it as a string. Depending on the type of data your custom precompile returns, you may encounter a similar issue. Let our pain be your gain!
+
+Note that in the case of a bad signature, the call to the precompile from Solidity may throw an error, or the precompile may return an X-Chain address that does not match the expected X-Chain address. Both responses should be considered negative. The only positive response is if the X-Chain address returned exactly matches the expected X-Chain address.
+
+### Modify the [example/params/config.go](./example/params/config.go) File
+
+As we mentioned earlier, the [example/params/config.go](./example/params/config.go) is used to add the chain configuration and managed the Subnet-evm via the genesis.json file.
+
+We have implemented the changes below in the [params.go](./example/precompile/params.go) file. At about [line 122](./example/params/config.go#L122) we have added the new custom contract `ContractXChainECRecoverConfig` in the ChainConfig struct with this we add or remove the precompiles through genesis.json file:
+
+```diff
+type ChainConfig struct {
+	ChainID *big.Int `json:"chainId"` // chainId identifies the current chain and is used for replay protection
+    .....
+    .....
++   ContractXChainECRecoverConfig	precompile.ContractXChainECRecoverConfig   `json:"contractXChainECRecover,omitempty"` 		  // Config for the contract XChain ECrecover
+}
+```
+
+Next, at about [line 563](./example/params/config.go#L563), we have included our custom precompiles intio the stateful precompiled configs:
+
+```diff
+func (c *ChainConfig) enabledStatefulPrecompiles() []precompile.StatefulPrecompileConfig {
+	statefulPrecompileConfigs := make([]precompile.StatefulPrecompileConfig, 0)
+
+	if c.ContractDeployerAllowListConfig.Timestamp() != nil {
+		statefulPrecompileConfigs = append(statefulPrecompileConfigs, &c.ContractDeployerAllowListConfig)
+	}
+    	.....
+    	.....
++   	if c.ContractXChainECRecoverConfig.Timestamp() != nil {
++ 		statefulPrecompileConfigs = append(statefulPrecompileConfigs, &c.ContractXChainECRecoverConfig)
++ 	}
+}
+```
```

Comment:
Are we able to put these additions in numerical order?

Is there an advantage to the current format?

Reply:
We don't understand your questions. Could you please elaborate as to specifically what you mean by "additions" and "numerical order" in this context?

Reply:
Hi @Julian-dev28 we've reordered the instructions by line number now thanks to your question. Should be easier to follow now.

Reply:
Thank you

It is

I'll run a few more reviews and let you know if I find anything else